src/lib.rs

```diff
@@ -132,9 +142,17 @@ impl Contract {
         }
     }
 
-    pub fn get_proposals(&self) -> Vec<VersionedProposal> {
+    pub fn get_proposals(&self, ids: Option< Vec<ProposalId> >) -> Vec<VersionedProposal> {
         near_sdk::log!("get_proposals");
-        self.proposals.to_vec()
+        if ids.is_some() {
+            let mut res: Vec<VersionedProposal> = vec![];
+            for id in ids.unwrap() {
+                res.push(self.proposals.get(id.into()).unwrap_or_else(|| panic!("Proposal id {} not found", id)));
+            }
+            res
+        } else {
```

Comment:
1. Use `if let Some(x) = x { ... x }` instead of `if x.is_some() { ... x.unwrap() }`
2. Let's use iterators
3. Let's not fail if there is no proposal for the given ID

```suggestion
        if let Some(ids) = ids {
            ids
                .into_iter()
                .filter_map(|id| self.proposals.get(id.into()))
                .collect()
        } else {
```